doc/sphinx/proof-engine/ltac.rst

```diff
@@ -997,98 +1181,42 @@ Pattern matching on terms: match
       and :n:`@ident` is present, it is assigned the "matched
       context", i.e. the initial term where the matched subterm is replaced by a
-      hole. The example below shows how to use such term contexts.
+      hole. The example below shows how to use such term contexts.  Note that `context`
+      (with very similar syntax) appearing after the `=>` is the :tacn:`context` tactic.
 
       For :tacn:`match` and :tacn:`multimatch`, if the evaluation of the :token:`ltac_expr`
       fails, the next matching subterm is tried. If no further subterm matches, the next branch
       is tried.  Matching subterms are considered from top to bottom and from left to
       right (with respect to the raw printing obtained by setting the
-      :flag:`Printing All` flag).
-
-     .. example:: Use of `context` in match on terms
-
-        .. coqtop:: all abort
-
-           Ltac f x :=
-             match x with
-               context f [S ?X] =>
-               idtac X;                    (* To display the evaluation order *)
-               assert (p := eq_refl 1 : X=1);    (* To filter the case X=1 *)
-               let x:= context f[O] in assert (x=O) (* To observe the context *)
-             end.
-           Goal True.
-           f (3+4).
+      :flag:`Printing All` flag).  :ref:`Example<match_term_context_ex>`.
 
    .. todo There's a more realistic example from @JasonGross here:
       https://github.com/coq/coq/pull/12103#discussion_r432996954
 
    .. todo @term after context is actually @lconstr_pattern (term with holes), which is spliced away
       Maybe worth restoring that distinction?
 
-   .. todo PR Didn't understand the following 2 paragraphs well enough to revise
-
-   Alternatively, when a metavariable in the form :n:`?id` occurs under binders,
-   say :n:`x__1, …, x__n` and the expression matches, the
-   metavariable is instantiated by a term which can then be used in any
-   context which also binds the variables :n:`x__1, …, x__n` with
-   same types. This provides with a primitive form of matching under
-   context which does not require manipulating a functional term.
+   .. note::
+      |Ltac| constructs are processed in two phases: an evaluation phase and an execution phase.
+      In most cases, tactics that may change the proof state are applied in the second phase.
 
-   If the match expression is in position to be applied to a goal
-   (e.g. it is not bound to a variable by a :n:`let in`), then this tactic is
-   applied. If the tactic succeeds, the list of resulting subgoals is the
-   result of the match expression. If the match expression is not in position
-   to be applied to a goal, then the result of the evaluation of :n:`@ltac_expr__1`
-   is the result of the match expression.
+      Unlike other tactics, `*match*` tactics get their first success (applying tactics to do
+      so) as part of the evaluation phase.  This can affect how early failures are processed
```

Comment:
This places too much weight on `assert_fails`.  This feature is quite useful independently from the `assert_*` tactics.  I'd suggest
```suggestion
      so) as part of the evaluation phase.  Among other things, this can affect how early failures are processed
```